File: xxf_arch/src/main/java/com/xxf/arch/core/SimpleActivityLifecycleCallbacks.java
package com.xxf.arch.core;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import java.util.Objects;

/**
 * @author xuanyouwu@163.com
 * @version 2.3.1
 * @Description 为Activity, fragment查找提供帮助
 * @date createTime：2018/9/7
 */
public class SimpleActivityLifecycleCallbacks implements Application.ActivityLifecycleCallbacks {

    public SimpleActivityLifecycleCallbacks() {
    }

    public void register(Application application) {
        Objects.requireNonNull(application);
        application.unregisterActivityLifecycleCallbacks(this);
        application.registerActivityLifecycleCallbacks(this);
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {

    }

    @Override
    public void onActivityStarted(Activity activity) {

    }

    @Override
    public void onActivityResumed(Activity activity) {

    }

    @Override
    public void onActivityPaused(Activity activity) {

    }

    @Override
    public void onActivityStopped(Activity activity) {

    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(Activity activity) {

    }
}

File: lib_application/src/main/java/com/xxf/application/activity/SimpleActivityLifecycleCallbacks.kt
package com.xxf.application.activity

import android.app.Activity
import android.app.Application
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle
import java.util.*

/**
 * @Author: XGod  xuanyouwu@163.com  17611639080  https://github.com/NBXXF     https://blog.csdn.net/axuanqq  xuanyouwu@163.com  17611639080  https://github.com/NBXXF     https://blog.csdn.net/axuanqq
 * @version 2.3.1
 * @Description 为Activity, fragment查找提供帮助
 * @date createTime：2018/9/7
 */
open class SimpleActivityLifecycleCallbacks : ActivityLifecycleCallbacks {
    open fun register(application: Application) {
        Objects.requireNonNull(application)
        application.unregisterActivityLifecycleCallbacks(this)
        application.registerActivityLifecycleCallbacks(this)
    }

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
    override fun onActivityStarted(activity: Activity) {}
    override fun onActivityResumed(activity: Activity) {}
    override fun onActivityPaused(activity: Activity) {}
    override fun onActivityStopped(activity: Activity) {}
    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
    }

    override fun onActivityDestroyed(activity: Activity) {}
}